module/VuFindApi/src/VuFindApi/Controller/SearchApiController.php

```diff
@@ -170,6 +177,16 @@ public function __construct(
                 $this->$key = $settings[$key];
             }
         }
+        $mainConfig = $this->getConfig('config');
+        $this->setApiKeyMode($mainConfig?->API_Keys?->mode ?? 'disabled');
+        if ($this->isApiKeyEnabled()) {
+            $this->setApiKeyHeader($mainConfig?->API_Keys?->header_field ?? 'X-API-KEY');
+            $this->setApiKeyService($this->getService(\VuFind\ApiKey\ApiKeyService::class));
+            if ($logRequests = $mainConfig?->API_Keys?->log_requests ?? false) {
+                $this->setApiKeyLogging($logRequests);
```

Comment:
Maybe worth an explicit cast here since the method expects a Boolean and the config probably returned some other type. (Not strictly necessary now but might matter in future if we enable strict types).

```suggestion
                $this->setApiKeyLogging((bool)$logRequests);
```